### PR TITLE
Optimized UriPath.java and created UriPath.test

### PR DESCRIPTION
**Pull request by thesnowgoose**
Moved @PrepareForTest to class header in WebSocketServletContextHandlerFactoryTest.java because running each test individually they failed. Each test should be independent of each other. 
Removed redundant casts.
Removed unnecessary variable in JettyServer.java

### CODE REVIEW

**Comment by thesnowgoose** [comment]
@perwendel @tipsy is there any chance you can check this pull request?? I think I did it good, and the UriPath class is gonna run faster.

**Comment by cesartg** [comment]
@thesnowgoose The UriPath code looks good! It is simpler and clearer that the copied code from Jetty. Also, you added the lacking unit test. Nice!

I see other files changes unrelated to the UriPath. I see they are minor improvements. They should be on other pull request but I am OK with them.

cc: @perwendel @tipsy

**Comment by tipsy** [comment]
Looks good to me, but @perwendel has to have a look (he might be on easter vacation now).

File: src/main/java/spark/resource/UriPath.java [hunk 1]
```diff
@@ -30,138 +20,43 @@
      * @return path or null.
      */
     public static String canonical(String path) {
-        if (path == null || path.length() == 0) {
+        if (path == null || path.isEmpty()){
```
Comment (dinesh707) [inline src/main/java/spark/resource/UriPath.java]: here you can use StringUtils (apache commons) "isBlank" or "isEmpty".

**Comment by thesnowgoose** [comment]
I did a minor change and improve the execution speed, also added another test case. @perwendel  It would be great if you could take a look.

**Comment by tipsy** [comment]
I like this PR. Could you update to follow the coding-conventions? https://github.com/perwendel/spark/blob/master/config/spark-formatter-intellij.jar

**Comment by tipsy** [comment]
Replaced by PR #792

**Pull request closed without merging**